.github/workflows/release.yml

```diff
@@ -0,0 +1,576 @@
+name: "Release Builds"
+on:
+  push:
+    branches:
+      - v[0-9].[0-9]**
+    tags:
+      - v[0-9].[0-9]**
+      - v[0-9].[0-9]**-**
+  workflow_dispatch:
+    inputs:
+      custom-ref:
+        required: false
+        description: "Input git-ref to build as a test release"
+
+env:
+  MESON_VERSION: '0.56.2'
+  EM_VERSION: '2.0.29'
+  EM_CACHE_FOLDER: 'emsdk'
+  TAISEI_NOPRELOAD: 0
+  TAISEI_PRELOAD_REQUIRED: 1
+
+jobs:
+  linux-release-build-x64:
+    name: "Linux (x64/Source)"
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Install Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
+
+    - name: Install Tools
+      run: >
+        sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+
+        sudo apt update || true # debian cache is not atomic and can sometimes fail
+
+        sudo apt install -y -qq
+        build-essential
+        libsdl2-dev
+        libx11-dev
+        libwayland-dev
+        python3-docutils
+        gnupg
+        gcc-10
+        g++-10
+        jq
+
+        sudo update-alternatives --install
+        /usr/bin/gcc gcc /usr/bin/gcc-10 60 --slave
+        /usr/bin/g++ g++ /usr/bin/g++-10
+
+        pip install
+        meson==${{ env.MESON_VERSION }}
+        ninja
+        zstandard
+        python-gnupg
+
+    - name: Import GPG Key
+      uses: taisei-project/ghaction-import-gpg@v5
+      with:
+        gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+        passphrase: ${{ secrets.GPG_PASSPHRASE }}
+
+    # NOTE: 'fetch-depth: 0' == "get all history", not "no history"
+    - name: Checkout Code
+      uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
+        fetch-depth: 1
+        ref: ${{ github.event.inputs.custom-ref }}
+
+    - name: Configure
+      run: >
+        git fetch --force --tags # see: https://github.com/actions/checkout/issues/290
+
+        meson setup build/linux
+        --native-file misc/ci/linux-x86_64-build-release.ini
+        --prefix=$(pwd)/build-release
+
+    - name: Set Package Version
+      run: |
+        if git describe --exact-match &>/dev/null && ${{ (github.event.inputs.custom-ref && false) || true }}
+        then
+            echo "BUILD_VERSION=${GITHUB_REF#refs/tags/v}" >> $GITHUB_ENV
+        else
+            echo "BUILD_VERSION=$(meson introspect --projectinfo build/linux | jq -r '.version' | cut -c 2-)" >> $GITHUB_ENV
+        fi
```

Comment:
(not a complete review yet)

It seems like the purpose of this variable is to "guess" the names of the release artifacts? I don't think this will work with custom refs, at least not in all cases. Regardless, we have a much better way of doing this already: just call `scripts/version.py`. That's what the build system uses to construct those file names.

Reply:
Done. Turns out `scripts/version.py` does do what I need. 

Currently macOS release builds are failing due to [this bug](https://github.com/actions/runner-images/issues/6507) but that should be fixed in the next day or two. I'll re-run [this build](https://github.com/StarWitch/taisei/actions/runs/3406325345/jobs/5664932541) to test it when it's merged.